diff -ur nvi-1.81.5-o/dist/configure.in nvi-1.81.5/dist/configure.in
--- nvi-1.81.5-o/dist/configure.in	2001-07-07 15:13:50.000000000 -0600
+++ nvi-1.81.5/dist/configure.in	2007-04-04 16:21:23.000000000 -0600
@@ -283,7 +283,7 @@
 *** is always available from ftp://ftp.gtk.org/.
 	  	])
 	    else
-		AM_PATH_GTK(1.2.0)
+		AM_PATH_GTK_2_0(2.0.0)
 		GTKINCS="$GTK_CFLAGS"
 		GTKLIBS="$GTK_LIBS $LIBS"
 	    fi
@@ -513,6 +513,15 @@
 	if test "$vi_cv_termlib" = no; then
 		AC_CHECK_LIB(termcap, tgetent,
 			[vi_cv_termlib=-ltermcap], [vi_cv_termlib=no])
+    if test "$vi_cv_termlib" = no; then
+      if test "$vi_cv_widechar" = yes; then
+				AC_CHECK_LIB(tinfow, tgetent,
+					[vi_cv_termlib=-ltinfow], [vi_cv_termlib=no])
+      else
+				AC_CHECK_LIB(tinfo, tgetent,
+					[vi_cv_termlib=-ltinfo], [vi_cv_termlib=no])
+			fi
+    fi
 	fi
 fi
 if test "$vi_cv_termlib" != no; then
@@ -900,15 +909,6 @@
 int fprintf ();
 ],[int i;], vi_cv_type_fprintf=int, vi_cv_type_fprintf=void)])
 
-AC_SUBST(NEED_FPRINTF_PROTO)
-AC_CACHE_CHECK([fprintf prototype needed], vi_cv_proto_fprintf,
-[AC_TRY_COMPILE([ #include <stdio.h>
-typedef int     (*funcPtr)();
-],[funcPtr ptr = (funcPtr) fprintf;], 
-vi_cv_proto_fprintf=no, 
-[vi_cv_proto_fprintf=yes, AC_DEFINE(NEED_FPRINTF_PROTO)])])
-
-
 dnl Check for DB 3
 AC_ARG_WITH(db3,
 	[  --with-db3=db3prefix    Path to db3 installation. ])
@@ -922,11 +922,14 @@
 	CFLAGS="-I$with_db3/include $CFLAGS"
 fi;
 
-AC_CHECK_LIB(db, db_create,
+saveLIBS="$LIBS"
+LIBS="$LIBS -ldb"
+AC_TRY_LINK([#include <db.h>], [db_create(0, 0, 0)],
 	[vi_cv_dbfatal="no"], [vi_cv_dbfatal="yes"])
 if test "$vi_cv_dbfatal" = "yes"; then
-	AC_MSG_ERROR([Need DB 3.])
+	AC_MSG_ERROR([Need DB 4.])
 fi
+LIBS="$saveLIBS"
 
 LDFLAGS="$saveLDFLAGS"
 
diff -ur nvi-1.81.5-o/dist/Makefile.am nvi-1.81.5/dist/Makefile.am
--- nvi-1.81.5-o/dist/Makefile.am	2001-07-16 04:49:10.000000000 -0600
+++ nvi-1.81.5/dist/Makefile.am	2007-04-04 16:17:56.000000000 -0600
@@ -238,7 +238,7 @@
 vi_motif_CFLAGS = @XINCS@
 
 VI=-DVI=\"$(bindir)/`echo vi-ipc | sed '$(transform)'`\"
-CFLAGS=	 -D__REGEX_PRIVATE $(OPTFLAG) @CFLAGS@ -I. -I$(visrcdir)/include \
+AM_CFLAGS=	 -D__REGEX_PRIVATE $(OPTFLAG) @CFLAGS@ -I. -I$(visrcdir)/include \
 	    @CPPFLAGS@ $(VI)
 PERL=	@vi_cv_path_perl@
 PERLLIB=@vi_cv_perllib@
