diff -ur nvi-1.81.5-o/dist/configure.in nvi-1.81.5/dist/configure.in
--- nvi-1.81.5-o/dist/configure.in	2001-07-07 15:13:50.000000000 -0600
+++ nvi-1.81.5/dist/configure.in	2007-04-04 16:21:23.000000000 -0600
@@ -283,7 +283,7 @@
 *** is always available from ftp://ftp.gtk.org/.
 	  	])
 	    else
-		AM_PATH_GTK(1.2.0)
+		AM_PATH_GTK_2_0(2.0.0)
 		GTKINCS="$GTK_CFLAGS"
 		GTKLIBS="$GTK_LIBS $LIBS"
 	    fi
@@ -513,6 +513,15 @@
 	if test "$vi_cv_termlib" = no; then
 		AC_CHECK_LIB(termcap, tgetent,
 			[vi_cv_termlib=-ltermcap], [vi_cv_termlib=no])
+    if test "$vi_cv_termlib" = no; then
+      if test "$vi_cv_widechar" = yes; then
+				AC_CHECK_LIB(tinfow, tgetent,
+					[vi_cv_termlib=-ltinfow], [vi_cv_termlib=no])
+      else
+				AC_CHECK_LIB(tinfo, tgetent,
+					[vi_cv_termlib=-ltinfo], [vi_cv_termlib=no])
+			fi
+    fi
 	fi
 fi
 if test "$vi_cv_termlib" != no; then
