diff -aurN nvi-1.79/build/Makefile.in nvi-1.79-new/build/Makefile.in
--- nvi-1.79/build/Makefile.in	Wed Oct 23 15:43:38 1996
+++ nvi-1.79-new/build/Makefile.in	Thu Apr  8 13:18:13 1999
@@ -95,15 +95,6 @@
 	    ($(mkdir) $(mandir)/cat1 && $(chmod) $(dmode) $(mandir)/cat1)
 	@echo "Installing man pages: $(mandir) ..."
 	cd $(mandir)/cat1 && $(rm) -f `echo vi.0 | sed '$(transform)'`
-	$(cp) $(srcdir)/docs/USD.doc/vi.man/vi.0 \
-	    $(mandir)/cat1/`echo vi.0 | sed '$(transform)'`
-	cd $(mandir)/cat1 && $(chmod) $(fmode) `echo vi.0 | sed '$(transform)'`
-	cd $(mandir)/cat1 && $(rm) -f `echo ex.0 | sed '$(transform)'`
-	cd $(mandir)/cat1 && $(rm) -f `echo view.0 | sed '$(transform)'`
-	cd $(mandir)/cat1 && $(ln) \
-	    `echo vi.0 | sed '$(transform)'` `echo ex.0 | sed '$(transform)'`
-	cd $(mandir)/cat1 && $(ln) \
-	    `echo vi.0 | sed '$(transform)'` `echo view.0 | sed '$(transform)'`
 	[ -d $(mandir)/man1 ] || \
 	    ($(mkdir) $(mandir)/man1 && $(chmod) $(dmode) $(mandir)/man1)
 	cd $(mandir)/man1 && $(rm) -f `echo vi.1 | sed '$(transform)'`
@@ -137,16 +137,6 @@
 	    $(chmod) $(dmode) $(datadir)/vi/catalog
 	(cd $(srcdir)/catalog && $(cp) $(cat) $(datadir)/vi/catalog && \
 	    cd $(datadir)/vi/catalog && $(chmod) $(fmode) *)
-	@echo "Installing Perl scripts: $(datadir)/vi/perl ..."
-	$(mkdir) $(datadir)/vi/perl && $(chmod) $(dmode) $(datadir)/vi/perl
-	[ -f VI.pm ] && $(cp) VI.pm $(datadir)/vi/perl && \
-	    cd $(datadir)/vi/perl && $(chmod) $(fmode) VI.pm)
-	(cd $(srcdir)/perl_scripts && $(cp) *.pl $(datadir)/vi/perl && \
-	    cd $(datadir)/vi/perl && $(chmod) $(fmode) *.pl)
-	@echo "Installing Tcl scripts: $(datadir)/vi/tcl ..."
-	$(mkdir) $(datadir)/vi/tcl && $(chmod) $(dmode) $(datadir)/vi/tcl
-	(cd $(srcdir)/tcl_scripts && $(cp) *.tcl $(datadir)/vi/tcl && \
-	    cd $(datadir)/vi/tcl && $(chmod) $(fmode) *.tcl)
 	@echo "Installing recover script: $(datadir)/vi/recover ..."
 	($(cp) recover $(datadir)/vi/recover && \
 	    $(chmod) $(emode) $(datadir)/vi/recover)
diff -aurN nvi-1.79/build/configure nvi-1.79-new/build/configure
--- nvi-1.79/build/configure	Wed Oct 23 15:53:07 1996
+++ nvi-1.79-new/build/configure	Thu Apr  8 13:18:13 1999
@@ -690,6 +690,7 @@
 	   OPTFLAG=${OPTFLAG-"-O2"};;
 irix6*)	   OPTFLAG=${OPTFLAG-"-O2"};;
 irix*)	   OPTFLAG=${OPTFLAG-"-O2"};;
+*linux)    CC=gcc;;
 lynxos*)   # Extract the first word of "gcc", so it can be a program name with args.
 set dummy gcc; ac_word=$2
 echo $ac_n "checking for $ac_word""... $ac_c" 1>&6
@@ -1736,7 +1737,7 @@
 	CPPFLAGS="-I\$(srcdir)/curses $CPPFLAGS"
 	cobjs="\$(COBJS)";;
 "other curses")
-	LIBS="-lcurses $LIBS";;
+	LIBS="-lncurses $LIBS";;
 esac
 
 echo $ac_n "checking for sys/mman.h""... $ac_c" 1>&6
diff -aurN nvi-1.79/build/recover.in nvi-1.79-new/build/recover.in
--- nvi-1.79/build/recover.in	Fri Oct 11 03:41:51 1996
+++ nvi-1.79-new/build/recover.in	Thu Apr  8 13:18:13 1999
@@ -7,8 +7,6 @@
 RECDIR="@vi_cv_path_preserve@"
 SENDMAIL="@vi_cv_path_sendmail@"
 
-echo 'Recovering nvi editor sessions.'
-
 # Check editor backup files.
 vibackup=`echo $RECDIR/vi.*`
 if [ "$vibackup" != "$RECDIR/vi.*" ]; then
diff -aurN nvi-1.79/ex/ex.c nvi-1.79-new/ex/ex.c
--- nvi-1.79/ex/ex.c	Fri Oct 11 00:49:46 1996
+++ nvi-1.79-new/ex/ex.c	Thu Apr  8 13:18:13 1999
@@ -2307,7 +2307,7 @@
 	 * underlying file, that's the real problem.
 	 */
 	if (sp->ep == NULL) {
-		ex_emsg(sp, cp->name, EXM_NOFILEYET);
+ 		ex_emsg(sp, cp ? cp->name : NULL, EXM_NOFILEYET);
 		return;
 	}
 
diff -aurN nvi-1.79/ex/ex_script.c nvi-1.79-new/ex/ex_script.c
--- nvi-1.79/ex/ex_script.c	Wed Sep 25 02:51:12 1996
+++ nvi-1.79-new/ex/ex_script.c	Thu Apr  8 13:18:13 1999
@@ -23,7 +23,8 @@
 #include <sys/select.h>
 #endif
 #include <sys/stat.h>
-#ifdef HAVE_SYS5_PTY
+/* glibc2.1 defines grantpt but there is no stropts.h */
+#if defined(HAVE_SYS5_PTY) && !defined(__GLIBC__)  
 #include <sys/stropts.h>
 #endif
 #include <sys/time.h>
@@ -625,7 +626,7 @@
 	F_CLR(gp, G_SCRWIN);
 }
 
-#ifdef HAVE_SYS5_PTY
+#if defined(HAVE_SYS5_PTY) && !defined(__GLIBC__)
 static int ptys_open __P((int, char *));
 static int ptym_open __P((char *));
 
diff -aurN nvi-1.79/regex/regcomp.c nvi-1.79-new/regex/regcomp.c
--- nvi-1.79/regex/regcomp.c	Sat Mar 19 17:21:28 1994
+++ nvi-1.79-new/regex/regcomp.c	Thu Apr  8 13:18:13 1999
@@ -609,7 +609,8 @@
 		REQUIRE(starordinary, REG_BADRPT);
 		/* FALLTHROUGH */
 	default:
-		ordinary(p, c &~ BACKSL);
+	  /* ordinary(p, c &~ BACKSL); -- Fix potential overflow */
+	        ordinary(p, (char)c);
 		break;
 	}
 
